src/core/stdc/string.d

```diff
@@ -29,80 +29,80 @@ nothrow:
 @nogc:
 
 ///
-pure void* memchr(in void* s, int c, size_t n);
+pure void* memchr(return const void* s, int c, size_t n);
 ///
-pure int   memcmp(in void* s1, in void* s2, size_t n);
+pure int   memcmp(scope const void* s1, scope const void* s2, size_t n);
 ///
-pure void* memcpy(void* s1, in void* s2, size_t n);
+pure void* memcpy(return void* s1, scope const void* s2, size_t n);
 version (Windows)
 {
     ///
-    int memicmp(in char* s1, in char* s2, size_t n);
+    int memicmp(scope const char* s1, scope const char* s2, size_t n);
 }
 ///
-pure void* memmove(void* s1, in void* s2, size_t n);
+pure void* memmove(return void* s1, scope const void* s2, size_t n);
 ///
-pure void* memset(void* s, int c, size_t n);
+pure void* memset(return void* s, int c, size_t n);
 
 ///
-pure char*  strcpy(char* s1, in char* s2);
+pure char*  strcpy(return char* s1, scope const char* s2);
 ///
-pure char*  strncpy(char* s1, in char* s2, size_t n);
+pure char*  strncpy(return char* s1, scope const char* s2, size_t n);
 ///
-pure char*  strcat(char* s1, in char* s2);
+pure char*  strcat(return char* s1, scope const char* s2);
 ///
-pure char*  strncat(char* s1, in char* s2, size_t n);
+pure char*  strncat(return char* s1, scope const char* s2, size_t n);
 ///
-pure int    strcmp(in char* s1, in char* s2);
+pure int    strcmp(scope const char* s1, scope const char* s2);
 ///
-int    strcoll(in char* s1, in char* s2);
+int    strcoll(scope const char* s1, scope const char* s2);
 ///
-pure int    strncmp(in char* s1, in char* s2, size_t n);
+pure int    strncmp(scope const char* s1, scope const char* s2, size_t n);
 ///
-size_t strxfrm(char* s1, in char* s2, size_t n);
+size_t strxfrm(scope char* s1, scope const char* s2, size_t n);
 ///
-pure char*  strchr(in char* s, int c);
+pure char*  strchr(return char* s, int c);
```

Comment:
This should also use `inout`, because the mutability of the return type should depend on the mutability if `s`. It's nice that a problem that C struggled for so much time (for [reference](https://www.lysator.liu.se/c/dmr-on-noalias.html)) with can be solved in D so elegantly.